site_scons/site_tools/doneapi.py

```diff
@@ -0,0 +1,83 @@
+"""SCons.Tool.doneapi
+
+Hack to support oneapi version of Intel compilers
+
+"""
+import sys, os
+
+is_linux = sys.platform.startswith('linux')
+
+if is_linux:
+    import SCons.Tool.gcc
+import SCons.Util
+import SCons.Warnings
+import SCons.Errors
+
+if not is_linux:
+    raise SCons.Errors.UserError("This tool doesn't work on non-Linux platforms")
+
+class OneapiError(SCons.Errors.InternalError):
+    pass
+
+class DetectCompiler(object):
```

Comment:
(pylint-missing-class-docstring) Missing class docstring
(pylint-useless-object-inheritance) Class 'DetectCompiler' inherits from object, can be safely removed from bases in python3
(pylint-too-few-public-methods) Too few public methods (1/2)